go/extractor/BUILD.bazel

```diff
@@ -0,0 +1,55 @@
+load("@rules_go//go:def.bzl", "go_library")
+load("@rules_pkg//pkg:mappings.bzl", "pkg_files")
+
+# gazelle:prefix github.com/github/codeql-go/extractor
+# gazelle:map_kind go_binary codeql_go_binary //go:rules.bzl
+
+go_library(
+    name = "extractor",
+    srcs = [
+        "extractor.go",
+        "gomodextractor.go",
+        "semaphore.go",
+    ],
+    importpath = "github.com/github/codeql-go/extractor",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//go/extractor/dbscheme",
+        "//go/extractor/diagnostics",
+        "//go/extractor/srcarchive",
+        "//go/extractor/trap",
+        "//go/extractor/util",
+        "//go/extractor/vendor/golang.org/x/mod/modfile",
+        "//go/extractor/vendor/golang.org/x/tools/go/packages",
+    ],
+)
```

Comment:
Question: it looks like this isn't generated by Gazelle. I assume we will have to update this by hand if we need to add more source files or dependencies?

Reply:
As I mentioned in another comment: gazelle actually updates `BUILD` files rather than generating them from scratch. So this target will be kept up-to-date by it, while the rest of the `BUILD` file will be left unchanged. But it's probably a good idea to be explicit about that in a comment here.

Reply:
Ah that's awesome, thank you for clarifying!